src/views/tutorial-view/index.tsx

```diff
@@ -191,26 +192,50 @@ function TutorialView({
 	]
 
 	/**
-	 * Keep track of progress for authenticated users
+	 * Set up variables for the tutorialId and collectionId, we use these below.
+	 * TODO: maybe this isn't necessary, I found it helpful for clarity.
+	 */
+	const tutorialId = id
+	const collectionId = collectionCtx.current.id
+	const collectionTutorialIds = collectionCtx.current.tutorials.map(
+		(t: TutorialLite) => t.id
+	)
+
+	/**
+	 * Prime `tutorial` and `collection` progress queries with a batch query.
+	 *
+	 * This should ideally include all `tutorial` and `collection` entries
+	 * we expect to render on the page, so that we only make one progress request.
+	 */
+	useProgressBatchQuery({
```

Comment:
This view probably merits some cleanup as it's getting chonky....  maybe even some grouping of all `progress` stuff into one hook (`useTutorialViewProgress`)? For now, focus was on "make it work". Very open to ideas here 👍